#! /usr/bin/env python

PACKAGE='jsk_perception'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("threshold1",   double_t, 0, "first threshold for the hysteresis procedure.",  100, 0, 500)
gen.add("threshold2",   double_t, 0, "second threshold for the hysteresis procedure.", 200, 0, 500)
gen.add("apertureSize", int_t,    0, "aperture size for the Sobel() operator.", 3, 1, 10)

gen.add("L2gradient",   bool_t,   0, " flag, indicating whether a more accurate  L_2 norm should be used to calculate the image gradient magnitude ( L2gradient=true ), or whether the default  L_1 norm is enough ( L2gradient=false ).", False)

exit(gen.generate(PACKAGE, "rectangle_detector", "RectangleDetector"))
